Realm/RLMRealm.mm

```diff
@@ -706,6 +710,72 @@ - (void)cancelWriteTransaction {
     }
 }
 
+#ifdef REALM_ASYNC_WRITES
+
+- (BOOL)inAsyncWriteTransaction {
+    return _realm->is_in_async_transaction();
+}
+
+- (void)setAsyncErrorHandler:(nullable RLMRealmAsyncErrorHandler)block {
+    if (block == nil) {
+        _realm->set_async_error_handler(nil);
+    }
+    else {
+        _realm->set_async_error_handler(^(RLMAsyncTransactionId asyncTransactionId, std::exception_ptr err) {
+            @autoreleasepool {
+                try {
+                    rethrow_exception(err);
+                } catch (...) {
+                    NSError *error = nil;
+                    RLMRealmTranslateException(&error);
+                    block(asyncTransactionId, error);
+                }
+            }
+        });
+    }
+}
+
+- (RLMAsyncTransactionId)beginAsyncWriteTransaction:(void(^)())block {
+    return _realm->async_begin_transaction(block);
```

Comment:
These can all throw exceptions and need to translate the exceptions from c++ to obj-c.